### PR TITLE
JAMES-2399 findMailboxWithPathLike should merge both DAOs

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java [hunk 1]
```diff
@@ -312,4 +316,17 @@ public void hasChildrenShouldReturnChildWhenExistsInV2Table() {
     
         assertThat(hasChildren).isTrue();
     }
+
+    @Test
+    public void findMailboxWithPathLikeShouldUnionBothVersions() {
```
Comment (rouazana) [inline mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java]: can you also add a test for hasChildren please?

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java [hunk 2]
```diff
@@ -159,11 +159,13 @@ public Mailbox findMailboxById(MailboxId id) throws MailboxException {
...
     public List<Mailbox> findMailboxWithPathLike(MailboxPath path) {
-        List<Mailbox> mailboxes = toMailboxes(path, mailboxPathV2DAO.listUserMailboxes(path.getNamespace(), path.getUser()));
-        if (mailboxes.isEmpty()) {
-            return toMailboxes(path, mailboxPathDAO.listUserMailboxes(path.getNamespace(), path.getUser()));
-        }
-        return mailboxes;
+        List<Mailbox> mailboxesV2 = toMailboxes(path, mailboxPathV2DAO.listUserMailboxes(path.getNamespace(), path.getUser()));
+        List<Mailbox> mailboxesV1 = toMailboxes(path, mailboxPathDAO.listUserMailboxes(path.getNamespace(), path.getUser()));
+
+        return ImmutableList.<Mailbox>builder()
```
Comment (mbaechler) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: is there any case where we have duplicates ?

File: mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java (hunk 1 above)
Comment (chibenwa) [reply mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapperTest.java]: hasChildrenShouldReturnChildWhenExistsInBothTables do test this, and we have tests with the children being in V1, V2 so I'd consider this tested.

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java (hunk 2 above)
Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: Possibly...

If so, priority should be given to V2.

Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java]: Fixed

**Comment by aduprat** [comment]
merged

**Pull request closed without merging**